### PR TITLE
[mobile]fix: added Carousel Edge Cases from Improvements & Refactoring

### PR DESCRIPTION
**Pull request by ppopendo**

### CODE REVIEW

File: tests/vodqa/test_carousel_edge_cases.py [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+        Test scenario:
+        1. Navigate to first carousel item (1 / 3) by swiping right multiple times
+        2. Attempt to swipe right again (past first item)
+        3. Verify carousel remains at first position
+
+        Expected:
+            - Carousel ID remains "1 / 3" after attempting to swipe past first item
+        """
+        # Arrange - ensure we're at the first item by swiping right until we can't
+        for _ in range(5):  # Max attempts to reach first item
```
Comment (ppopendo) [inline tests/vodqa/test_carousel_edge_cases.py]: @copilot 
please remove loops and conditional statements from the entire file

**Comment by ppopendo** [comment]
@copilot open a new pull request to apply changes based on the comments in [this thread](https://github.com/ppopendo/mobile_automation_learning/pull/57#pullrequestreview-3801800012)

**Comment by ppopendo** [comment]
@copilot 
both tests ended with an error, details below
```
test_swipe_right_at_first_item_stays_at_first:
Expected :{'final_position': '1 / 3', 'initial_position': '1 / 3'}
Actual   :{'final_position': '1 / 3', 'initial_position': '2 / 3'}

test_swipe_left_at_last_item_stays_at_last
Expected :{'final_position': '3 / 3', 'initial_position': '3 / 3'}
Actual   :{'final_position': '1 / 3', 'initial_position': '1 / 3'}
```

**Comment by ppopendo** [comment]
@copilot 
despite merged PR60  -> both tests ended with an error, 

```
test:   test_swipe_right_at_first_item_stays_at_first: 
result:
Expected :{'final_position': '1 / 3', 'initial_position': '1 / 3'}
Actual   :{'final_position': '2 / 3', 'initial_position': '1 / 3'}

test:  test_swipe_left_at_last_item_stays_at_last:
result: 
Expected :{'final_position': '3 / 3', 'initial_position': '3 / 3'}
Actual   :{'final_position': '1 / 3', 'initial_position': '1 / 3'}
```

**Comment by ppopendo** [comment]
@copilot 
despite merged PR61 -> on test case ended with an error,
```
test: test_swipe_left_at_first_item_stays_at_first ✔️ 
test: test_swipe_right_at_last_item_stays_at_last
result: 
Expected :{'final_position': '3 / 3', 'initial_position': '3 / 3'}
Actual   :{'final_position': '1 / 3', 'initial_position': '2 / 3'}
```

**Comment by ppopendo** [comment]
@copilot 
please improve below test cases:
all tests should be stable
```
test: test_swipe_right_at_first_item_stays_at_first
result:
  Common items:
  {'initial_position': '1 / 3'}
  Differing items:
  {'final_position': '2 / 3'} != {'final_position': '1 / 3'}
  
  Full diff:
    {
  -     'final_position': '1 / 3',
  ?                        ^
  +     'final_position': '2 / 3',
  ?                        ^
        'initial_position': '1 / 3',
    }

assert {'initial_position': '1 / 3', 'final_position': '2 / 3'} == {'initial_position': '1 / 3', 'final_position': '1 / 3'}


test: test_swipe_left_at_last_item_stays_at_last
result: 
assert {'initial_position': '1 / 3', 'final_position': '1 / 3'} == {'initial_position': '3 / 3', 'final_position': '3 / 3'}
  
  Differing items:
  {'final_position': '1 / 3'} != {'final_position': '3 / 3'}
  {'initial_position': '1 / 3'} != {'initial_position': '3 / 3'}
  
  Full diff:
    {
  -     'final_position': '3 / 3',
  ?                        ^
  +     'final_position': '1 / 3',
  ?                        ^
  -     'initial_position': '3 / 3',
  ?                          ^
  +     'initial_position': '1 / 3',
  ?                          ^
    }

test: test_fling_returns_false_at_first_boundary
result: ok

test: test_fling_returns_false_at_last_boundary
result:
assert True is False

test: test_multiple_swipes_right_at_first_position_stays_stable
result: 
E       AssertionError: Carousel should remain at first position after multiple swipes: {'all_positions_at_first': False, 'positions': ['2 / 3', '1 / 3', '2 / 3']}
E       assert {'all_positions_at_first': False, 'positions': ['2 / 3', '1 / 3', '2 / 3']} == {'all_positions_at_first': True, 'positions': ['1 / 3', '1 / 3', '1 / 3']}
E         
E         Differing items:
E         {'all_positions_at_first': False} != {'all_positions_at_first': True}
E         {'positions': ['2 / 3', '1 / 3', '2 / 3']} != {'positions': ['1 / 3', '1 / 3', '1 / 3']}
E         
E         Full diff:
E           {
E         -     'all_positions_at_first': True,
E         ?                               ^^^
E         +     'all_positions_at_first': False,
E         ?                               ^^^^
E               'positions': [
E         +         '2 / 3',
E                   '1 / 3',
E         -         '1 / 3',
E         ?          ^
E         +         '2 / 3',
E         ?          ^
E         -         '1 / 3',
E               ],
E           }


test: test_multiple_swipes_left_at_last_position_stays_stable
result:
assert {'all_positions_at_last': False, 'positions': ['1 / 3', '1 / 3', '1 / 3']} == {'all_positions_at_last': True, 'positions': ['3 / 3', '3 / 3', '3 / 3']}
  
  Differing items:
  {'all_positions_at_last': False} != {'all_positions_at_last': True}
  {'positions': ['1 / 3', '1 / 3', '1 / 3']} != {'positions': ['3 / 3', '3 / 3', '3 / 3']}
  
  Full diff:
    {
  -     'all_positions_at_last': True,
  ?                              ^^^
  +     'all_positions_at_last': False,
  ?                              ^^^^
        'positions': [
  -         '3 / 3',
  ?          ^
  +         '1 / 3',
  ?          ^
  -         '3 / 3',
  ?          ^
  +         '1 / 3',
  ?          ^
  -         '3 / 3',
  ?          ^
  +         '1 / 3',
  ?          ^
        ],
    }

test: test_full_carousel_traversal 
result: 
assert {'start_position': '1 / 3', 'forward_visited_all': False, 'backward_visited_all': False, 'ended_at_first': True} == {'start_position': '1 / 3', 'forward_visited_all': True, 'backward_visited_all': True, 'ended_at_first': True}
  
  Common items:
  {'ended_at_first': True, 'start_position': '1 / 3'}
  Differing items:
  {'backward_visited_all': False} != {'backward_visited_all': True}
  {'forward_visited_all': False} != {'forward_visited_all': True}
  
  Full diff:
    {
  -     'backward_visited_all': True,
  ?                             ^^^
  +     'backward_visited_all': False,
  ?                             ^^^^
        'ended_at_first': True,
  -     'forward_visited_all': True,
  ?                            ^^^
  +     'forward_visited_all': False,
  ?                            ^^^^
        'start_position': '1 / 3',
    }



```

**Comment by ppopendo** [comment]
@copilot open a new pull request to apply changes based on the comments in [this thread](https://github.com/ppopendo/mobile_automation_learning/pull/57#pullrequestreview-3835440777)
